### PR TITLE
Skip pagination parameters when set to null and add infinityModelFailed event.

### PR DESCRIPTION
**Pull request by ShogunPanda**
Hi.
In my server (which is based on cursor pagination) I have to make sure no extra parameters are set.
Therefore I try to set `perPageParam` and `pageParam` to `null`. 

Unfortunately I noticed the parameters still get added like this:
```
$URL?...&null=2&...
```

I write a patch to completely exclude parameters when the names are not set.
Hope you like it!
  Paolo

### CODE REVIEW

**Comment by davidgoli** [comment]
@ShogunPanda I support this PR. However, the tests are failing.

**Comment by davidgoli** [comment]
To be fair, it looks like the same failure as other recent builds. But a test to confirm this behavior would be nice. Thanks!

**Comment by ShogunPanda** [comment]
@davidgoli Ok, will write a test soon!

**Comment by ShogunPanda** [comment]
I couldn't write a test yet.
Instead, I added a `infinityModelFailed` error so we can handle error instead of having an unhandled exception.

File: README.md [hunk 1]
```diff
@@ -273,6 +273,13 @@ Triggered on the route when the infinityModel is fully loaded.
 
 * infinityModel
 
+**infinityModelFailed**
+
+Triggered on the route whenever server return an error.
```
Comment (hhff) [inline README.md]: Grammar:

`Triggered on the route whenever the server returns an error.`

**Comment by hhff** [comment]
Tests are passing now - but echo'ing @davidgoli some tests would be great

File: README.md (hunk 1 above)
Comment (ShogunPanda) [inline README.md]: Good point. I'm embarrassed now. Will fix it as well.

Comment (hhff) [inline README.md]: haha don't be embarrassed!  Code reviews are for picking up little things :+1:

**Comment by mdentremont** [comment]
:+1: for `infinityModelFailed` :)

**Comment by davidgoli** [comment]
I'd recommend removing `infinityModelFailed` due to the fix introduced in https://github.com/hhff/ember-infinity/pull/125

**Comment by davidgoli** [comment]
also, please merge in master

**Comment by hhff** [comment]
@ShogunPanda - can you please merge in master?

**Comment by ShogunPanda** [comment]
Sure, will take care of this ASAP

**Comment by ShogunPanda** [comment]
I closed this in favor of a new clean PR. See #147.

**Pull request closed without merging**